### ISSUE TITLE
Having trouble finding a distro for my RPi2

### ISSUE THREAD

**Issue by texadactyl** [texadactyl/rpi_clock#21]
@tehmessiah75 @vacumcollapse @DEV-XiA @MRDGH2821 @youanan

I have a spare Raspberry Pi 2 model B.  And, I finally found stock of the Adafruit PiTFT 3.5" Touch Screen for Raspberry Pi at Tinkersphere.  My issue is finding a new or old Raspbian O/S image with a desktop that will work with the TFT product.  Hopefully, I did not receive a lemon as it is currently not functional beyond showing a blank powered-up screen.

In the "Easy Install" instructions at Adafruit (https://learn.adafruit.com/adafruit-pitft-3-dot-5-touch-screen-for-raspberry-pi/easy-install-2), they are recommending distros that are "Lite" i.e. without a desktop.  I can do that and add a desktop or simply add enough X-Windows to make it work.  But, that complicates the prep notes for a non-programming user.

You lot have gotten rpi_clock to work on Raspberry Pis so I have a few questions.  There are no wrong answers!  Opinions are most welcome.  Any other issue readers are also solicited.

1. Are you using the Adafruit PiTFT 3.5" Touch Screen? 
2. Which Raspbian or Raspberry Pi OS with desktop are you using?  A precise URL for download would be useful.
3. If you downloaded a Lite O/S, how did you add enough X-Windows to satisfy Tk so that rpi_clock ran?  Was this straight-forward enough for a non-programming user?
4. What did you have to do in raspi-config to get it to work?  Enable SSH, of course, for remote connection.  How about SPI?
5. Did you hand-edit the boot config.txt file?  If so, details, please.
6. Any other comments?  Anything else that might help a non-programming user?

By the way, you can run with an ```ssh -Y``` connection in v5.0 (ready for installation).

**Comment by vacuumcollapse** [comment]
Hi Richard

Before diving into the detail, just checking that you’re not stuck in the infamous fkms problem caused by the latest RpiOS (which is linux bullseye)?  If you have an HDMI screen older than 5yrs or so, the pi will boot but no hdmi output occurs – you need to go into config.txt on the sdcard and change the dtoverlay line that refers to kms to fkms, then reboot and it should all work from there…

Insanely avoidable and widespread problem caused by linux bullseye kernel change, my coder mates tell me.

Matt

From: Richard Elkins ***@***.***>
Sent: 16 January 2023 11:45
To: texadactyl/rpi_clock ***@***.***>
Cc: Matt Bidwell ***@***.***>; Mention ***@***.***>
Subject: [texadactyl/rpi_clock] Having trouble finding a distro for my RPi2 (Issue #21)


@tehmessiah75<https://github.com/tehmessiah75> @vacumcollapse<https://github.com/vacumcollapse>

I have a spare Raspberry Pi 2 model B. And, I finally found stock of the Adafruit PiTFT 3.5" Touch Screen for Raspberry Pi at Tinkersphere. My issue is finding a new or old Raspbian O/S image with a desktop that will work with the TFT product. Hopefully, I did not receive a lemon as it is currently not functional beyond showing a blank powered-up screen.

In the "Easy Install" instructions at Adafruit (https://learn.adafruit.com/adafruit-pitft-3-dot-5-touch-screen-for-raspberry-pi/easy-install-2), they are recommending distros that are "Lite" i.e. without a desktop. I can do that and add a desktop or simply add enough X-Windows to make it work. But, that complicates the prep notes for a non-programming user.

You two have gotten rpi_clock to work on Raspberry Pis so I have a few questions. There are no wrong answers! Opinions are most welcome.

  1.  Are you using the Adafruit PiTFT 3.5" Touch Screen?
  2.  Which Raspbian or Raspberry Pi OS with desktop are you using? A precise URL for download would be useful.
  3.  If you downloaded a Lite O/S, how did you add enough X-Windows to satisfy Tk so that rpi_clock ran? Was this straight-forward enough for a non-programming user?
  4.  What did you have to do in raspi-config to get it to work? Enable SSH, of course, for remote connection. How about SPI?
  5.  Did you hand-edit the boot config.txt file? If so, details, please.
  6.  Any other comments? Anything else that might help a non-programming user?

By the way, you can run with an ssh -Y connection in v5.0 (ready for installation).

**Comment by texadactyl** [comment]
@vacumcollapse  Thanks for the response.  I am not having any issues with HDMI output regardless of which O/S I use.  My current dilemma centers around the PiTFT screen which never shows any output.  It seems to be electronically healthy because it does light up when power is applied.

**Comment by vacuumcollapse** [comment]
OK, so answers:

1 yes that’s the screen I’m using
2 I’m using an RPI3 with current full 32bit RPi os, incl. desktop, installed using the RPI installer from the raspberry pi website.  Critically, I used the FBCP script on the Adafruit page, rather than the ‘console mode’ script.
3 n/a
4 I have SPI and I2C enabled; can’t remember which one it uses
5 config.txt has the following uncommented which may be relevant (can’t remember if I changed them or the adafruit script did): dtparam=i2c_arm=on and dtparam=spi=on, and it also has the following at the bottom:
# --- added by adafruit-pitft-helper Thu Dec 22 23:30:21 2022 ---
[all]
hdmi_force_hotplug=1  # required for cases when HDMI is not plugged in!
dtparam=spi=on
dtparam=i2c1=on
dtparam=i2c_arm=on
dtoverlay=pitft35-resistive,rotate=90,speed=20000000,fps=20
# --- end adafruit-pitft-helper Thu Dec 22 23:30:21 2022 ---

6 I can’t think of anything else, but will keep trying to remember.  As ever, if it just worked, it wasn’t memorable!

Matt

**Comment by texadactyl** [comment]
Thanks, Matt.  Yes, that is what I did too in one of my tries.
It looks like my PiTFT is, unfortunately, DOA.  )-:
Returning it to Tinkersphere.

**Comment by MRDGH2821** [comment]
I personally used this screen - https://www.uctronics.com/display/uctronics-3-5-inch-touch-screen-raspberry-hdmi-tft-lcd-mini-display.html 
(This seems to be an update of what I had. I did get a stylus & 3 heat sinks)

And I had used Raspbian OS (Bullseye, from the official installer, full version) with SSH enabled, no SPI.
I did this in 2019. Later on I formatted the SD card for several reasons, so I do not remember any more details.

It was a headache to install display Drivers as everything was 1024x768, being squeezed inside 400x300 screen

Apart from that this was able to satisfy my requirements:
1. Auto sync time when internet was connected (this is from OS side)
2. Black background with Red colour font is what I had done for my usage (this was to mimic the digital clock which was sent for repairs)
3. Auto start on boot up

I hope this helps. 
(On a side note, I'm wondering how did you findout that I used this repo? 🤔 )

**Comment by texadactyl** [comment]
@MRDGH2821 You created issue #9.

**Comment by tehmessiah75** [comment]
> Are you using the Adafruit PiTFT 3.5" Touch Screen?

No, I bought a Cheap touchscreen from eBay. [3.5 Inch LCD TFT Touch Screen Display Monitor for Raspberry pi 2/3 Model B](https://www.ebay.com.au/itm/373686635653)

> Which Raspbian or Raspberry Pi OS with desktop are you using?  A precise URL for download would be useful.

I used the [Raspberry Pi Imager](https://www.raspberrypi.com/software/) I used the default Raspberry Pi image suitable for RPi2B (that's what I have). The only things I changed was in the Advanced Options 1. I set a Hostname, 2. enabled SSH for username and password 3. Set username and password. Also once I had set up this Weather Clock and my Pi-Hole I used raspi-config and went into performance options and turned on Read Only File System in an effort to extend the life of the SD card. (Do this after you have configured everything and tested to confirm it works)

> If you downloaded a Lite O/S, how did you add enough X-Windows to satisfy Tk so that rpi_clock ran?  Was this straight-forward enough for a non-programming user?

I initially tried the Lite OS and had so many issues (plus a suspected dying SD card Still not sure on that though because tests show its fine). I started with Lite and then installed XFCE desktop but its just not the same as installing an OS with the GUI as part of the install. I could get it to manually start but I couldn't get it to auto start at boot. Once I installed the default RPi OS everything worked as designed.

> What did you have to do in raspi-config to get it to work?  Enable SSH, of course, for remote connection.  How about SPI?

 
I didn't have to do anything to get it to work. It just did work by following the instructions. I didn't even do the TFT touch calibration.

> Did you hand-edit the boot config.txt file?  If so, details, please.

No.

> Any other comments?  Anything else that might help a non-programming user?

Just use the Raspberry Pi Imager and use the default OS suitable for your Pi and everything should work as designed.
